### ISSUE TITLE
Right-to-left languages 

### ISSUE THREAD

**Issue by Fakeel** [jemmybutton/byrne-euclid#65]
I have started a challenge to translate the whole book in Arabic. I am not an expert in Context but I am learning. 

As you know some languages writing system like Arabic, Hebrew, Pashto, Persian, Urdu, and Sindhi are written from right to left. So I solve most of issue for the Arabic language, only one remain issue.

Solving the Lettrine placement for both pages

What should I do?

[page_design.zip](https://github.com/jemmybutton/byrne-euclid/files/5804948/page_design.zip)

**Comment by jemmybutton** [comment]
@Fakeel Wow! That's super cool!

There was a bug in my initials placement algorithm when an RtL script is used. Now it's [fixed](https://github.com/jemmybutton/byrne-euclid/commit/fa5b304036bd6f8670695136ff1b2fe893cf5851) (hopefully) and your example seems to work ok. All you need to do now is to update `preamble_be.tex` file.

![byrne_ar](https://user-images.githubusercontent.com/7447349/104477620-628f8c80-55d2-11eb-9fe0-cdfc94e912e3.png)

There most likely are still some problems left, so please don't hesitate to ask, if something goes wrong, i'll try to fix it.
One thing that definitely doesn't work now is `lettrines.mp` script for the generated initials with ornaments. It currently only supports one glyph at a time, but you seem to use full words. I'll try and fix that at some point. 

As a side note: are there initials in Arabic typography? Maybe you have some references?

**Comment by Fakeel** [comment]
There aren't initials in Arabic typography and it does not use block letters and separate letters. 
Most of the Arabic art uses calligraphy design.

In this project the whole Arabic word should be place inside the lettrine or use calligraphy design like the one below

https://www.arabiccalligraphygenerator.com/
![image](https://user-images.githubusercontent.com/1329740/104506517-8c5aaa80-55f6-11eb-8fa9-eba3cec7b521.png)

**Comment by jemmybutton** [comment]
If initials do not belong in Arabic typography, I can simply add an option to turn them off.

**Comment by Fakeel** [comment]
I have tried your [fixed](https://github.com/jemmybutton/byrne-euclid/commit/fa5b304036bd6f8670695136ff1b2fe893cf5851)
Nothing is chanced

I am missing something?

[page_design.zip](https://github.com/jemmybutton/byrne-euclid/files/5811438/page_design.zip)
[page_design.pdf](https://github.com/jemmybutton/byrne-euclid/files/5811435/page_design.pdf)

**Comment by jemmybutton** [comment]
Strange, my output seems ok for the same file.
[page_design.pdf](https://github.com/jemmybutton/byrne-euclid/files/5811448/page_design.pdf)
What version of ConTeXt are you using?

**Comment by Fakeel** [comment]
ConTeXt version 2018.04.04 00:51

mtx-context     | ConTeXt Process Management 1.02
mtx-context     |
mtx-context     | main context file: /usr/share/texmf/tex/context/base/mkiv/context.mkiv
mtx-context     | current version: 2018.04.04 00:51

**Comment by jemmybutton** [comment]
I have a newer version and it apparently is different from the older ones (I had to use newer system on TravisCI, because build failed with the older one). Currently the code decides where to place the initial using an internal variable `\textdirection` which seems to have appeared only in recent versions. So I suspect, this may be the problem. You can try, as a a quick fix, setting it manually somewhere in the preamble (say, right before `starttext` like so:

```
\newcount\textdirection
\textdirection=1 %1 stands for RtL
```

Or update to a newer version.

**Comment by Fakeel** [comment]
Thank, it works.

It will be helpful if there is a note or comment to indicate which version of ConTeXt should be used.

**Comment by jemmybutton** [comment]
Great!
There's a nasty [bug](https://github.com/jemmybutton/byrne-euclid/issues/62) in the version I'm currently using, so I can't recommend it, at the same time, everything seemed to be reasonably backwards-compatible till this case. Once i update, i'll indicate the best version to use in the `README.md`.

**Comment by jemmybutton** [comment]
@Fakeel Multi-glyph initials, as you do in Arabic, should work [now](https://github.com/jemmybutton/byrne-euclid/commit/8a3ddbd0fcff9a15aa3bf0120af9edc27b64a529), but only in theory. In practice, the function which produces text outlines in ConTeXt's MetaPost currently [doesn't respect RTL text](http://tracker.luatex.org/view.php?id=1023). Also some longer words may not fit well into the square box and I'm thinking about making this box's width variable.

**Comment by Fakeel** [comment]
You are right, ConTeXt's MetaPost currently doesn't respect RtL text as shown in  
[out.pdf](https://github.com/jemmybutton/byrne-euclid/files/5889692/out.pdf)

Also, don't worry about box's width because I am trying to choose Arabic word that don't exceed 3 letter.

**Comment by jemmybutton** [comment]
@Fakeel With [this change](https://github.com/jemmybutton/byrne-euclid/commit/d1d27049b2cddb3f5422a2ad8a954832146616b1) initials' placeholders should work again for RtL, but you still can't generate ornaments for them. The bug with RtL is [fixed](http://tracker.luatex.org/view.php?id=1023#c1732) in future versions of ConTeXt, so I guess, I'd just wait for the fixed version to be released and then everything should work fine.
